llvm/lib/IR/RepoTicket.cpp

```diff
@@ -118,25 +127,46 @@ static GODigestState accumulateGOFinalDigestOrInitialDigestAndGetGODigestState(
 
 // Update the GO's hash value by adding the hash of its dependents.
 template <typename FunctionType>
-static bool updateDigestUseDependenciesAndContributions(
-    const GlobalObject *GO, MD5 &GOHash, GOStateMap &Visited, GOInfoMap &GOIMap,
-    FunctionType AccumulateGODigest) {
+static auto updateDigestUseDependenciesAndContributions(
+    const GlobalObject *GO, GOStateMap &Visited, GOInfoMap &GOIMap,
+    MemoizedHashes &GOHashCache, FunctionType AccumulateGODigest)
+    -> std::tuple<bool, size_t, DigestType> {
   if (GO->isDeclaration())
-    return false;
+    return std::make_tuple(false, std::numeric_limits<size_t>::max(),
+                           NullDigest);
+
+  const auto NumVisited = Visited.size();
+  LLVM_DEBUG(dbgs() << "Computing hash for \"" << GO->getName() << "\" (#"
+                    << NumVisited << ")\n");
+
+  // If the hash has been memorized, we can return the result immediately.
```

Comment:
I think it's best to stick to the term memoized. memorized is close but not as specific.

Reply:
Fixed, thanks.